### DISCUSSION TITLE
Like .use functions as a prepend is there any way to add a .postpend?

### DISCUSSION THREAD

**Discussion by rcuching** [mswjs/msw#2155] [Q&A] [answered]
Relatively new to MSW so I may just get this flat out wrong in terms of trying to use the tool. So I figured id ask to try and get some context on why im wrong if I am or if the improvement could be useful.

I have a msw that is set up before a test runs that returns `a` and url `/` for example. In my test I have a mutation that changes the return value so now I want `a'` to be returned from that point on and I do a refetch at `/`. I have this test that runs the mutation in a nested describe suite. So im using server.use() to try and add additional logic that isnt needed in the other tests. Should i just set an individual server.use in every nested describe instead? 

Since I think if .use is functioning as a prepend and MSW goes through the handlers right to left. if i put the `a'` handler in the use. that will be the handler called first and not the initial handler thats used to set up the server which i expect to run first.

**Comment by kettanaito** [comment, marked as answer]
Hi, @rcuching. That's a good question. 

`.use()` always prepends request handlers because the order of handlers [matters](https://mswjs.io/docs/concepts/request-handler#execution-order). There is no way to "push" handlers (add to the end) and I don't believe there's ever a reason to do so. 

> Should i just set an individual server.use in every nested describe instead?

You have a few options here. 

## Option 1: `server.use()` in individual `describe()` blocks

```js
describe('1', () => {
  describe('1.1', () => {
    beforeAll(() => server.use(...overrides))
    afterAll(() => server.resetHandlers())
  })

  describe('1.2', () => {
    beforeAll(() => server.use(...overrides))
    afterAll(() => server.resetHandlers())
  })
})
```

Don't forget to reset the overrides you add with `server.resetHandlers()`.

## Option 2: `server.boundary()`

If your test suite is rather nested and you don't want to accidentally forget to call `.use()` or `.resetHandlers()`, consider using [`server.boundary()`](https://mswjs.io/docs/api/setup-server/boundary).

```js
describe('1', () => {
  describe('1.1', () => {
    it('1.1.1', server.boundary(() => {
      server.use(...overrides)
    }))
  })

  describe('1.2', () => {
    it('1.2.1', server.boundary(() => {
      server.use(...overrides)
    }))
  })
})
```

`server.boundary()` wrapper will prevent any handler overrides to escape that test's scope. You don't have to reset any handlers either because nothing leaves the scope of the boundary. I have the feeling you will benefit more from `server.boundary()` in your case but I will leave that decision up to you. 

## Related materials

- [Network behavior overrides](https://mswjs.io/docs/best-practices/network-behavior-overrides)

**Reply by rcuching** [reply]
Thank you! ill try to put these into use
